src/python/py_image_generation_pipelines.cpp

```diff
@@ -68,7 +53,70 @@ auto text2image_generate_docstring = R"(
     :rtype: ov.Tensor
 )";
 
+// Trampoline class to support inheritance from Generator in Python
+class PyGenerator : public ov::genai::Generator {
+public:
+    float next() override {
+        PYBIND11_OVERRIDE_PURE(float, Generator, next);
+    }
 
+    ov::Tensor randn_tensor(const ov::Shape& shape) override {
+        PYBIND11_OVERRIDE(ov::Tensor, Generator, randn_tensor, shape);
+    }
+};
+
+py::list to_py_list(const ov::Shape shape) {
+    py::list py_shape;
+    for (auto d : shape)
+        py_shape.append(d);
+
+    return py_shape;
+}
+
+class TorchGenerator : public ov::genai::CppStdGenerator {
+    py::module_ m_torch;
+    py::object m_torch_generator, m_float32;
+    py::object m_torch_tensor; // we need to hold torch.Tensor to avoid memory destruction
+public:
+    explicit TorchGenerator(uint32_t seed) : CppStdGenerator(seed) {
+        try {
+            m_torch = py::module_::import("torch");
+            m_torch_generator = m_torch.attr("Generator")("device"_a="cpu").attr("manual_seed")(seed);
+            m_float32 = m_torch.attr("float32");
+        } catch (const py::error_already_set& e) {
+            if (e.matches(PyExc_ModuleNotFoundError)) {
+                PyErr_WarnEx(PyExc_ImportWarning, "'torch' module is not installed. Random generation will fall back to 'CppStdGenerator'", 1);
+            } else {
+                // Re-throw other exceptions
+                throw;
+            }
+        }
+    }
+
+    float next() override {
+        if (m_torch) {
+            return m_torch.attr("randn")(1, "generator"_a=m_torch_generator, "dtype"_a=m_float32).attr("item")().cast<float>();
+        } else {
+            return CppStdGenerator::next();
+        }
+    }
+
+    ov::Tensor randn_tensor(const ov::Shape& shape) override {
+        if (m_torch) {
+            m_torch_tensor = m_torch.attr("randn")(to_py_list(shape), "generator"_a=m_torch_generator, "dtype"_a=m_float32);
+            py::object numpy_tensor = m_torch_tensor.attr("numpy")();
+            py::array numpy_array = py::cast<py::array>(numpy_tensor);
+
+            if (!numpy_array.dtype().is(py::dtype::of<float>())) {
+                throw std::runtime_error("Expected a NumPy array with dtype float32");
+            }
+
+            return ov::Tensor(ov::element::f32, shape, numpy_array.mutable_data());
```

Comment:
So you keep `m_torch_tensor` as a member to keep this returned tensor valid. Calling `randn_tensor()` again would override `m_torch_tensor` and loose the previous data. You probably need that trick with an allocator here.

Reply:
Good point, updated